modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -39,5 +41,12 @@ object ConstructionService {
       .map(TransactionIdentifier(_))
       .attemptT
       .leftMap(_ => MalformedTransaction)
+
+    def getAccountIdentifiers(operations: List[Operation]): Option[NonEmptyList[AccountIdentifier]] = {
+      val accountIdentifiers =
+        operations.filter(_.amount.exists(_.value.value.value < 0)).flatMap(_.account)
+
+      Option.when(accountIdentifiers.nonEmpty)(NonEmptyList.fromList(accountIdentifiers)).flatten
```

Comment:
if you decided to use `NonEmptyList.fromList` it already returns an Option.
None in case the list is empty and Some in case there are elements, so there is no need to wrap it additionally into second Option and flatten instantly.

so it should be just:
```scala
NonEmptyList.fromList(accountIdentifiers)
```

Reply:
Changed.